### DISCUSSION TITLE
How to import global types/interfaces?

### DISCUSSION THREAD

**Discussion by autsada** [alephjs/aleph.js#129] [Q&A] [answered]
Hi,

I am playing around with aleph quite a bit to learn it, and have one question.
 
My question is how to import global types/interfaces that globally available in react (@types/react (global.d.ts)), for example, "HTMLButtonElement".

**Comment by shadowtime2000** [comment]
Please don't use issues for questions, use discussions or the discord instead.

**Reply by autsada** [reply]
Noted, thank you.

**Comment by ije** [comment]
please try `import type {HTMLButtonElement} from "react"`

**Reply by shadowtime2000** [reply]
I think same thing from `@types/react` may also work.

**Reply by autsada** [reply]
Hi @ije 

It doesn't work, still got an error `".....deno/deps/https/cdn.esm.sh/e862842f481428a7127d8f058cabba75c67c60146de500c2378dfaedaed1b1a0"' has no exported member 'HTMLButtonElement'."`

**Reply by autsada** [reply]
Hi @shadowtime2000 

I also tried to import "@types/react" but go this error ```
"error: Import 'https://esm.sh/@types/react@17.0.1' failed: 500 Internal Server Error
exit with code: 1"
```

**Reply by shadowtime2000** [reply]
@autsada Oh yeah forgot about that, I don't think esm.sh allows importing `@types/` packages.

**Reply by shadowtime2000** [reply]
I think its possible its available in the global scope because of the TS JSX stuff but not sure...

**Reply by autsada** [reply]
I saw in [dext](https://deno.land/x/dext@0.10.3) the preact framework in deno that is also inspired by next, they have tsconfig.json that set lib to "dom" so it might enable us to use that global types but not sure about this.
```
{
  "compilerOptions": {
    "lib": ["esnext", "dom", "deno.ns"],
    "jsx": "react",
    "jsxFactory": "h",
    "jsxFragmentFactory": "Fragment"
  }
}
```
And also this [issue](https://github.com/denoland/deno/issues/8070) in deno that might relate to the issue I am facing.

**Reply by autsada** [reply]
By the way, is it possible to use tsconfig.json in aleph?

**Comment by ije** [comment]
since we switchy to swc that doesn't support type check yet instead of tsc, tsconfig doesn't work in alephjs.

**Comment by ije** [comment, marked as answer]
`/// <reference lib="dom" />` can work

**Reply by autsada** [reply]
Thank you, I will give it a try.

**Reply by autsada** [reply]
Hi @ije 

Yes, it works.
Just one more question please, where is the right file to put this directive? Now I have it in the root file app.tsx

**Reply by shadowtime2000** [reply]
@autsada I think you will have to put it in all the files that are using it...

**Reply by autsada** [reply]
@shadowtime2000 thank you.
